dev-tools/packaging/templates/linux/postinstall.sh.tmpl

```diff
@@ -47,5 +17,9 @@ fi
 echo "create symlink "$symlink" to "$new_agent_dir/elastic-agent""
 ln -s "$new_agent_dir/elastic-agent" "$symlink"
 
+# reload systemctl and then restart service
+echo "systemd enable/restart elastic-agent"
 systemctl daemon-reload 2> /dev/null
+systemctl enable elatic-agent 2> /dev/null || true
```

Comment:
Does this cause any documentation changes for `delayed enrollment` feature?  I'm not sure, but that is the one place I can think of where we might have accidentally relied on the service not starting immediately.

Reply:
Hmm, that is very true.

One of the reasons I liked this change was because upon fresh install the service was started, then enrollment worked correctly. With it off and enrollment the daemon is not running and it errors on restarting the Elastic Agent. Another reason is for upgrading, where it does restart the service and it becomes the new version.

This does cause an issue with delay enrollment though. I guess we just need to make a decision on the path we want to take.

Reply:
I have been thinking about this more and with delayed enrollment this actually isn't too much of an issue. If you install the package and then enroll with delayed enrollment then it will not restart the Elastic Agent so the running agent will be unaffected. It would only take affect if you restarted the service, at that point it would enroll.

I think I am in favor of keeping this new behavior and maybe add a note in the delayed enrollment documentation. It running will not affect it, so I think this is less of a worry.

Reply:
I have a preference for the new behavior as well.  It is more inline with https://www.debian.org/doc/debian-policy/index.html

I just wanted to make sure we didn't have any surprises with delayed enrollment.